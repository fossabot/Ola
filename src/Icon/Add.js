import React from 'react'

const Add = () => <path
  d="M14,2 C7.38326923,2 2,7.38326923 2,14 C2,20.6167308 7.38326923,26 14,26 C20.6167308,26 26,20.6167308 26,14 C26,7.38326923 20.6167308,2 14,2 Z M20.0018344,15.0161123 L15.0608255,15.0161123 L15.0608255,19.9803097 L13.0231809,19.9803097 L13.0231809,15.0161123 L8.01169143,15.0161123 L8.01169143,12.9724987 L13.0231809,12.9724987 L13.0231809,8.02789951 L15.0608255,8.02789951 L15.0608255,12.9724987 L20.0018344,12.9724987 L20.0018344,15.0161123 Z" />

export default Add