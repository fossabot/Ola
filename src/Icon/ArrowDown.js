import React from 'react'

const ArrowDown = () => <path
    d="M14 26l10.706-9.292a1 1 0 00-1.318-1.5L15 22.5V3a1 1 0 00-1.993-.117L13 22.5l-8.292-7.206c-.36-.361-1.037-.406-1.408-.02s-.396.945-.091 1.338L14 26z" />

export default ArrowDown
