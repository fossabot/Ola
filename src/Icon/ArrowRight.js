import React from 'react'

const ArrowRight = () => <path
  d="M26 14L16.708 3.293a1 1 0 00-1.5 1.319L22.5 13H3a1 1 0 00-.117 1.993L22.5 15l-7.207 8.292c-.36.36-.405 1.036-.02 1.408.386.37.946.395 1.339.09L26 14z" />

export default ArrowRight