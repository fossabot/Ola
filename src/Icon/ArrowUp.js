import React from 'react'

const ArrowUp = () => <path
  d="M14 2l10.706 9.292a1 1 0 01-1.318 1.5L15 5.5V25a1 1 0 01-1.993.117L13 5.5l-8.292 7.206c-.36.361-1.037.406-1.408.02s-.396-.945-.091-1.338L14 2z" />

export default ArrowUp
