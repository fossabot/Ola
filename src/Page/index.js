import React from 'react'
import {default as PT} from 'prop-types'
import cx from 'classnames'

const Page = ({ title, url, icon, status, className, link, ...props }) => {
  const styles = cx(
    'ola_page',
    { 'ola-skeleton': status !== 'loaded' },
    { 'is-loading': status === 'loading' },
    className
  )

  const pageUrl = link ?
    <a href={link} rel="noopener noreferrer" target="_blank" className="ola_page-url">{url}</a> :
    <span className="ola_page-url">{ url }</span>

  return (
    <p className={styles} {...props}>
      { status === 'loaded'
        ? <>
          { icon &&
            <span className="ola_page-icon">
              { icon }
            </span>
          }
          <strong className="ola_page-title">{ title }</strong>
          { pageUrl }
        </>
        : <>
          { icon && <span className="ola_page-icon"></span> }
          <strong className="ola_page-title"></strong>
          <span className="ola_page-url"></span>
        </>
      }
    </p>
  )
}

Page.defaultProps = {
  icon: null,
  link: null,
  status: 'loaded',
  className: null
}

Page.propTypes = {
  icon: PT.element,
  /** Page Title*/
  title: PT.string.isRequired,
  /** Page Url */
  url: PT.string.isRequired,
  /** Url link */
  link: PT.string,
  /** Page status */
  status: PT.oneOf(['loaded', 'loading', 'empty']),
  /** Extra className */
  className: PT.string,
}

export default Page
