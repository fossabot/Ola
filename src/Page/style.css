.ola_page {
    display: grid;
    grid-template-columns: auto 1fr;
    grid-column-gap: var(--size-4);
}
.ola_page-title {
    --skeleton-width: 20em;
    font: var(--font-callout);
    font-weight: var(--bold);
    display: block;
    grid-column: 2;
    grid-row: 1;
}
.ola_page-url {
    --skeleton-width: 14em;
    font: var(--font-caption);
    display: block;
    grid-column: 2;
    grid-row: 2;
    justify-self: start;
    white-space: nowrap;
    max-width: 100%;
    overflow: hidden;
    text-overflow: ellipsis;
}
span.ola_page-url {
    color: var(--gray);
}
.ola_page-icon {
    grid-column: 1;
    grid-row: 1 / 3;
    max-width: var(--size-8);
    align-self: center;
    color: var(--gray-light);
}
.ola-skeleton {
    & .ola_page-icon {
        --skeleton-width: var(--size-8);

        &:before {
            height: var(--skeleton-width);
        }
    }
}