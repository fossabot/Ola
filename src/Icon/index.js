import React from 'react'
import {default as PT} from 'prop-types'
import cx from 'classnames'

// Icons
import ChevronDown from './ChevronDown'
import ChevronUp from './ChevronUp'
import ChevronLeft from './ChevronLeft'
import ChevronRight from './ChevronRight'
import Clock from './Clock'
import Close from './Close'
import Error from './Error'
import Help from './Help'
import Home from './Home'
import Page from './Page'
import Search from './Search'
import Success from './Success'
import User from './User'
import Warning from './Warning'

const ICONS = {
  'chevronDown': ChevronDown,
  'chevronUp': ChevronUp,
  'chevronLeft': ChevronLeft,
  'chevronRight': ChevronRight,
  'clock': Clock,
  'close': Close,
  'error': Error,
  'help': Help,
  'home': Home,
  'page': Page,
  'search': Search,
  'success': Success,
  'user': User,
  'warning': Warning
}

const Icon = ({ name, size, extraClass, ...props }) => {
  const SpecificIcon = ICONS[name]
  const IconClasses = cx('ola_icon', `is-${size}`, extraClass)
  return (
    <svg width="28" height="28" viewBox="0 0 28 28" fillRule="evenodd" className={IconClasses} {...props}>
      <SpecificIcon />
    </svg>
  )
}

Icon.defaultProps = {
  name: 'help',
  size: 'medium',
  extraClass: null
}

Icon.propTypes = {
  /** Icons */
  name: PT.oneOf([
    'chevronDown',
    'chevronUp',
    'chevronLeft',
    'chevronRight',
    'clock',
    'close',
    'error',
    'help',
    'home',
    'page',
    'search',
    'success',
    'user',
    'warning'
  ]),
  /** Size */
  size: PT.oneOf(['small', 'medium', 'big']),
  /** Extra className */
  extraClass: PT.string
}

export default Icon
