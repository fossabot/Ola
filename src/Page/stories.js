import React from 'react'
import { storiesOf } from '@storybook/react'

import Page from './'
import Icon from '../Icon'
import Level from '../Level'

storiesOf('Page')
  .add('Default', () => (
    <figure><Page title="Pricing - marketgoo" url="/pricing/" /></figure>
  ))
  .add('With icon', () => (
    <figure><Page title="Easy SEO Tools - marketgoo" url="/index" icon={<Icon name="home"/>} /></figure>
  ))
  .add('With level and link', () => (
    <figure><Page title="MarkeGoo PRO - marketgoo" url="/marketgoo-pro/" link="http://google.com" icon={<Level variant="medium"/>} /></figure>
  ))