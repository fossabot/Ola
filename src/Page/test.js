import React from 'react'
import Page from './'
import Icon from '../Icon'
import renderer from 'react-test-renderer'

it('Default Page', () => {
  const tree = renderer
    .create(<Page title="Title test" url="test.url.com" />)
    .toJSON()
  expect(tree).toMatchSnapshot()
})

it('Page with link', () => {
  const tree = renderer
    .create(<Page title="Title test" url="test.url.com" link="link.url.com" />)
    .toJSON()
  expect(tree).toMatchSnapshot()
})

it('Page with Icon', () => {
  const tree = renderer
    .create(<Page title="Title test" url="test.url.com" icon={<Icon />} />)
    .toJSON()
  expect(tree).toMatchSnapshot()
})
