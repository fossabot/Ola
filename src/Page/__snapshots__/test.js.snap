// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Default Page 1`] = `
<p
  className="ola_page"
>
  <strong
    className="ola_page-title"
  >
    Title test
  </strong>
  <span
    className="ola_page-url"
  >
    test.url.com
  </span>
</p>
`;

exports[`Page with Icon 1`] = `
<p
  className="ola_page has-icon"
>
  <span
    className="ola_page-icon"
  >
    <svg
      className="ola_icon is-medium"
      fillRule="evenodd"
      height="28"
      viewBox="0 0 28 28"
      width="28"
    >
      <path
        d="M14 2C7.376 2 2 7.376 2 14s5.376 12 12 12 12-5.376 12-12S20.624 2 14 2zm1 19h-2v-2h2v2zm2.07-7.75l-.9.92C15.45 14.9 15 15.5 15 17h-2v-.5c0-1.1.45-2.1 1.17-2.83l1.24-1.26c.37-.36.59-.86.59-1.41 0-1.1-.9-2-2-2s-2 .9-2 2h-2c0-2.21 1.79-4 4-4s4 1.79 4 4c0 .88-.36 1.68-.93 2.25z"
      />
    </svg>
  </span>
  <strong
    className="ola_page-title"
  >
    Title test
  </strong>
  <span
    className="ola_page-url"
  >
    test.url.com
  </span>
</p>
`;

exports[`Page with link 1`] = `
<p
  className="ola_page"
>
  <strong
    className="ola_page-title"
  >
    Title test
  </strong>
  <a
    className="ola_page-url"
    href="link.url.com"
    rel="noopener noreferrer"
    target="_blank"
  >
    test.url.com
  </a>
</p>
`;
